packages/iov-ledger-bns/src/app.ts

```diff
@@ -11,20 +11,32 @@ const cmdPubkeyWithPath = 5;
 const cmdAppVersion = 0xca;
 
 export function getPublicKeyWithIndex(transport: TransportNodeHid, i: number): Promise<Uint8Array> {
-  const pathComponent = Slip10RawIndex.hardened(i).asNumber();
-  return sendChunks(transport, appCode, cmdPubkeyWithPath, encodeUint32(pathComponent));
+  const pathComponent = Slip10RawIndex.hardened(i);
+  const chunk = buildPathPrefix([pathComponent]);
+  return sendChunks(transport, appCode, cmdPubkeyWithPath, chunk);
 }
 
 export function signTransactionWithIndex(
   transport: TransportNodeHid,
   transaction: Uint8Array,
   i: number,
 ): Promise<Uint8Array> {
-  const pathComponent = Slip10RawIndex.hardened(i).asNumber();
-  const data = new Uint8Array([...encodeUint32(pathComponent), ...transaction]);
+  const pathComponent = Slip10RawIndex.hardened(i);
+  const data = new Uint8Array([...buildPathPrefix([pathComponent]), ...transaction]);
   return sendChunks(transport, appCode, cmdSignWithPath, data);
 }
 
+// construct a binary length-prefixed path to send to the ledger
+function buildPathPrefix(path: ReadonlyArray<Slip10RawIndex>): Uint8Array {
+  // I'm sure there is a better way to do this, but...
```

Comment:
haha, I see what you mean. I don't see an obvious way to improve and it looks correct.